dot/sync/syncer.go

```diff
@@ -119,25 +119,30 @@ func NewService(cfg *Config) (*Service, error) {
 
 // HandleBlockAnnounceHandshake handles a block that a peer claims to have through a HandleBlockAnnounceHandshake
 func (s *Service) HandleBlockAnnounceHandshake(blockNum *big.Int) *network.BlockRequestMessage {
-	if blockNum == nil || s.highestSeenBlock.Cmp(blockNum) != -1 {
+	bestNum, err := s.blockState.BestBlockNumber()
+	if err != nil {
+		s.logger.Error("failed to get best block number", "error", err)
+		return nil // TODO: handle this / panic?
+	}
+
+	if blockNum == nil || bestNum.Cmp(blockNum) != -1 || s.highestSeenBlock.Cmp(blockNum) >= 0 {
```

Comment:
What is this check assessing?

Reply:
if the block number passed in isn't higher than our current head of chain, or the "highest seen" block (ie highest block a peer claims to have) then return